### PR TITLE
[iOS] Support restoreState

### PR DESCRIPTION
**Pull request by TomoLV**

### CODE REVIEW

File: example/lib/devices_list/devices_bloc.dart [hunk 1]
```diff
@@ -48,7 +48,15 @@ class DevicesBloc {
     bleDevices.clear();
     Blemulator().addSimulatedPeripheral(SensorTag());
     Blemulator().simulate();
-    _bleManager.createClient()
+    _bleManager.createClient(
+          restoreStateIdentifier: "example-restore-state-identifier",
+          restoreStateAction: (peripherals) {
+            peripherals.forEach((peripheral) {
+              bleDevices.add(ConnectedBleDevice(peripheral.name, peripheral.identifier, peripheral));
```
Comment (mikolak) [inline example/lib/devices_list/devices_bloc.dart]: It should be as DisconnectedBleDevice

File: lib/internal/bridge/lib_core.dart [hunk 2]
```diff
@@ -31,7 +31,7 @@ class FlutterBleLib extends FlutterBLE
           if (jsonString == null)
             return null;
           else {
-            List<Map<String, dynamic>> restoredPeripheralsJson =
+            List<dynamic> restoredPeripheralsJson =
                 jsonDecode(jsonString);
```
Comment (mikolak) [inline lib/internal/bridge/lib_core.dart]: `List<Map<String, dynamic>> restoredPeripheralsJson =
            (jsonDecode(jsonString) as List<dynamic>).cast();`

Comment (TomoLV) [reply lib/internal/bridge/lib_core.dart]: Changed as requested

File: example/lib/devices_list/devices_bloc.dart (hunk 1 above)
Comment (TomoLV) [reply example/lib/devices_list/devices_bloc.dart]: Changed to BleDevice.notConnected

**Pull request merged**